cpp/src/jit/cache.cpp

```diff
@@ -112,9 +114,14 @@ jitify2::ProgramCache<>& get_program_cache(jitify2::PreprocessedProgramData prep
   auto existing_cache = caches.find(preprog.name());
 
   if (existing_cache == caches.end()) {
-    auto res = caches.insert(
-      {preprog.name(),
-       std::make_unique<jitify2::ProgramCache<>>(100, preprog, nullptr, get_program_cache_dir())});
+    auto max_files = std::getenv("LIBCUDF_KERNEL_CACHE_FILE_LIMIT");
+    auto res       = caches.insert({preprog.name(),
+                              std::make_unique<jitify2::ProgramCache<>>(
+                                100,
```

Comment:
Why not keep a lot in the memory too?

Reply:
I don't know how much memory a JIT'd kernel takes up, so I left this number unmodified.

Reply:
updated it to cache (effectively) unlimited in-memory and on-disk kernels, as was the behavior prior to jitify2.